pyomo/contrib/mindtpy/single_tree.py

```diff
@@ -0,0 +1,257 @@
+from __future__ import division
+
+
+from pyomo.core import Constraint, Expression, Objective, minimize, value, Var
+from pyomo.opt import TerminationCondition as tc
+from pyomo.contrib.mindtpy.nlp_solve import (solve_NLP_subproblem,
+                                             handle_NLP_subproblem_optimal, handle_NLP_subproblem_infeasible,
+                                             handle_NLP_subproblem_other_termination, solve_NLP_feas)
+from pyomo.contrib.gdpopt.util import copy_var_list_values, identify_variables
+from math import copysign
+from pyomo.environ import *
+from pyomo.core.expr import current as EXPR
+from math import fabs
+from pyomo.repn import generate_standard_repn
+import logging
+from pyomo.common.dependencies import attempt_import
+
+cplex, cplex_available = attempt_import('cplex')
+if cplex_available:
+    from cplex.callbacks import LazyConstraintCallback
+else:
+    logging.warning(
+        "Cplex python API is not found. Therefore, lp-nlp is not supported")
+    # Other solvers (e.g. Gurobi) are not supported yet
```

Comment:
Since `single_tree` is only imported conditionally if it is going to be used, I think you can just `import cplex` here.  In fact there is no reason not to: if `cplex` is not available, line 27 will fail with "`NameError: name 'LazyConstraintCallback' is not defined`".  I think it is better to get the `ImportError` from '`import cplex`...

Reply:
Thanks for the advice.